src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -0,0 +1,1351 @@
+    if (lastBitmap == null) {
+      // There is no value with same high: the rank is a sum of cardinalities
+      for (Entry<Integer, BitmapDataProvider> bitmap : highToBitmap.entrySet()) {
+        if (bitmap.getKey().intValue() > high) {
+          break;
+        } else {
+          result += bitmap.getValue().getLongCardinality();
+        }
+      }
+    } else {
+      for (BitmapDataProvider bitmap : highToBitmap.values()) {
+        if (bitmap == lastBitmap) {
+          result += bitmap.rankLong(low);
+          break;
+        } else {
+          result += bitmap.getLongCardinality();
+        }
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * 
+   * @param high for which high bucket should we compute the cardinality
+   * @return the highest validatedIndex
+   */
+  protected int ensureCumulatives(int high) {
+    if (allValid) {
+      // the whole array is valid (up-to its actual length, not its capacity)
+      return highToBitmap.size();
+    } else if (compare(high, firstHighNotValid) < 0) {
+      // The high is strictly below the first not valid: it is valid
+
+      // sortedHighs may have only a subset of valid values on the right. However, these invalid
+      // values have been set to maxValue, and we are here as high < firstHighNotValid ==> high <
+      // maxHigh()
+      int position = binarySearch(sortedHighs, high);
+
+      if (position >= 0) {
+        // This high has a bitmap: +1 as this index will be used as right (excluded) bound in a
+        // binary-search
+        return position + 1;
+      } else {
+        // This high has no bitmap: it could be between 2 highs with bitmaps
+        int insertionPosition = -position - 1;
+        return insertionPosition;
+      }
+    } else {
+      // For each deprecated buckets
+      SortedMap<Integer, BitmapDataProvider> tailMap =
+          highToBitmap.tailMap(firstHighNotValid, true);
+
+      // TODO .size on tailMap make an iterator: arg
+      int indexOk = highToBitmap.size() - tailMap.size();
+
+      Iterator<Entry<Integer, BitmapDataProvider>> it = tailMap.entrySet().iterator();
+      while (it.hasNext()) {
+        Entry<Integer, BitmapDataProvider> e = it.next();
+        int currentHigh = e.getKey();
+
+        if (compare(currentHigh, high) > 0) {
+          // No need to compute more than needed
+          break;
+        } else if (e.getValue().isEmpty()) {
+          // highToBitmap can not be modified as we iterate over it
+          if (latestAddedHigh != null && latestAddedHigh.getKey().intValue() == currentHigh) {
+            // Dismiss the cached bitmap as it is removed from the NavigableMap
+            latestAddedHigh = null;
+          }
+          it.remove();
+        } else {
+          ensureOne(e, currentHigh, indexOk);
+
+          // We have added one valid cardinality
+          indexOk++;
+        }
+
+      }
+
+      if (highToBitmap.isEmpty() || indexOk == highToBitmap.values().size()) {
```

Comment:
Do you really need to call `#size()` on the collection of values instead of calling it on `highToBitmap` ?

Reply:
Clearly useless